torchvision/transforms/v2/functional/_color.py

```diff
@@ -33,9 +33,14 @@ def rgb_to_grayscale(inpt: torch.Tensor, num_output_channels: int = 1) -> torch.
 def _rgb_to_grayscale_image(
     image: torch.Tensor, num_output_channels: int = 1, preserve_dtype: bool = True
 ) -> torch.Tensor:
-    if image.shape[-3] == 1:
+    if num_output_channels not in (1, 3):
+        raise ValueError(f"num_output_channels must be 1 or 3, got {num_output_channels}.")
+    if image.shape[-3] == 1 and num_output_channels == 1:
         return image.clone()
-
+    if image.shape[-3] == 1 and num_output_channels == 3:
+        s = [-1] * len(image.shape)
+        s[-3] = 3
+        image = image.expand(s)
```

Comment:
Should we simply do

```suggestion
        return image.expand(s)
```

here? The output should be the same and it would save some extra computation from `l_img = r.mul(0.2989).add_(g, alpha=0.587).add_(b, alpha=0.114)`. Not critical though.

Reply:
You are right. This saves some ops. Done.